packages/exhibitor-migration-script/extra/exhibitor-migration-script/bin/dcos-exhibitor-migrate-perform

```diff
@@ -0,0 +1,86 @@
+LOG_INDEX_DIR_GOAL = '/var/lib/dcos/exhibitor/zookeeper/transactions'
+
+
+class ValidationError(Exception):
+    def __init__(self, msg, exit_status=8):
+        self.msg = msg
+        self.exit_status = exit_status
+
+
+def get_config_json(response):
+    if response.status_code == 200 and response.headers['Content-Type'] == 'application/json':
+        return response.json()['config']
+    else:
+        raise ValidationError("Error reading configuration from Exhibitor", 2)
+
+
+def migration_already_complete(config):
+    if config['rollInProgress'] is True:
+        return False
+    else:
+        return (config['zookeeperDataDirectory'] == ZOOKEEPER_DATA_DIR_GOAL and
+                config['zookeeperLogDirectory'] == ZOOKEEPER_LOG_DIR_GOAL and
+                config['logIndexDirectory'] == LOG_INDEX_DIR_GOAL)
+
+
+def assert_can_migrate(config):
+    if config['rollInProgress'] is True:
+        raise ValidationError("Rolling upgrade in progress. {}".format(config['rollStatus']), 4)
+    else:
+        zk_data_directory = config['zookeeperDataDirectory']
+        if zk_data_directory != ZOOKEEPER_DATA_DIR_START:
+            raise ValidationError("ZooKeeper Data Directory does not match expected value. expected: {} actual: {}"
+                                  .format(ZOOKEEPER_DATA_DIR_START, zk_data_directory))
+        zk_log_directory = config['zookeeperLogDirectory']
+        if zk_log_directory != ZOOKEEPER_LOG_DIR_START:
+            raise ValidationError("ZooKeeper Log Directory does not match expected value. expected: {} actual: {}"
+                                  .format(ZOOKEEPER_LOG_DIR_START, zk_log_directory))
+        log_index_directory = config['logIndexDirectory']
+        if log_index_directory != LOG_INDEX_DIR_START:
+            raise ValidationError("Log Index Directory does not match expected value. expected: {} actual: {}"
+                                  .format(LOG_INDEX_DIR_GOAL, log_index_directory))
+
+
+def update_config(current_config):
+    config = copy.deepcopy(current_config)
+    config['zookeeperDataDirectory'] = ZOOKEEPER_DATA_DIR_GOAL
+    config['zookeeperLogDirectory'] = ZOOKEEPER_LOG_DIR_GOAL
+    config['logIndexDirectory'] = LOG_INDEX_DIR_GOAL
+    return config
+
+
+if __name__ == '__main__':
+    try:
+        resp = requests.get("http://localhost:8181/exhibitor/v1/config/get-state", timeout=(3, 10))
+        json = get_config_json(resp)
+        if migration_already_complete(json):
+            print("Migration already complete")
+            sys.exit(0)
+        else:
+            assert_can_migrate(json)
+            updated_config = update_config(json)
+            set_rolling_response = requests.post("http://localhost:8181/exhibitor/v1/config/set-rolling",
+                                                 json=updated_config, timeout=(3, 10))
+            if set_rolling_response.status_code == 200:
+                print("Rolling update started")
+                sys.exit(0)
+
```

Comment:
if set_rolling_response.status_code != 200 we silently fall out here and end up exiting 0 with no message printed.